lib/validation/validators.js

```diff
@@ -85,21 +85,27 @@ module.exports = Validators = {
   },
 
   // eslint-disable-next-line no-inline-comments, spaced-comment
-  before(value/*, [diff, unit][, diff, unit]*/) {
-    const args = [].slice.call(arguments, 1);
-    let date = moment(value, dateFormat);
-    let diff;
-    let unit;
-    while (args.length) {
-      diff = args.shift();
-      unit = args.shift() || 'years';
-      date = date.add(diff, unit);
+  before(value, date/*, [diff, unit][, diff, unit]*/) {
```

Comment:
In this context the square brackets are indicating optional arguments, so it's actually neither of the things you described.

The correct signature is:

```
before('2012-01-01', 5, 'years', 6, 'months');
```

That is, zero or more *pairs* of arguments can be provided.

Yes, this is misleading. No, I don't have a better suggestion.